tests/rptest/tests/audit_log_test.py

```diff
@@ -481,43 +475,39 @@ def __init__(self, logger, filter_fn, stop_cond, ocsf_server):
                 self.filter_fn = filter_fn
                 self.stop_cond = stop_cond
                 self.ocsf_server = ocsf_server
-                self.next_offset_ingest = 0
 
             def ingest(self, records):
-                new_records = records[self.next_offset_ingest:]
-                self.next_offset_ingest = len(records)
-                new_records = [json.loads(msg['value']) for msg in new_records]
+                new_records = [json.loads(msg['value']) for msg in records]
+                self.logger.info(f"Ingested: {len(new_records)} records")
```

Comment:
This changes the test behavior no?  Instead of just checking the new records, it checks _all_ records in the audit log.  I wrote my tests to expect the original behavior.

Reply:
It wasn't working that way though. It would start from offset 0 every time `read_all_from_audit_log` is called anyway because the offset that it saved to keep track of where it was wasn't persistently stored across invocations of the method.

Reply:
fair enough!